### PR TITLE
[2.x.x] Generate txhashset archives on 720 block intervals. 

### PR DESCRIPTION
**Pull request by quentinlesceller**
Replace https://github.com/mimblewimble/grin/pull/2813. Cause this previous PR was merged on master branch.

### CODE REVIEW

File: p2p/src/protocol.rs [hunk 1]
```diff
@@ -297,15 +299,17 @@ impl MessageHandler for Protocol {
 					sm_req.hash, sm_req.height
```
Comment (garyyu) [inline p2p/src/protocol.rs]: This log has been changed (in master before reverting) as:
```
				debug!(
					"handle_payload: txhashset request for {} at {}, response with {} at {}",
					sm_req.hash, sm_req.height, txhashset_header.height, txhashset_header_hash,
					"handle_payload: txhashset req for {} at {}",
					sm_req.hash, sm_req.height
				);
```

File: chain/tests/chain_test_helper.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+/// Mine a chain of specified length to assist with automated tests.
+/// Must call clean_output_dir at the end of your test.
+pub fn mine_chain(dir_name: &str, chain_length: u64) -> Chain {
+	global::set_mining_mode(ChainTypes::AutomatedTesting);
+
+	// add coinbase data from the dev genesis block
+	let mut genesis = genesis::genesis_dev();
+	let keychain = keychain::ExtKeychain::from_random_seed(false).unwrap();
+	let key_id = keychain::ExtKeychain::derive_key_id(0, 1, 0, 0, 0);
+	let reward = reward::output(&keychain, &key_id, 0, false).unwrap();
```
Comment (garyyu) [inline chain/tests/chain_test_helper.rs]: Ah, some interfaces has been changed and the test code need a bit rework now:

```
error[E0061]: this function takes 5 parameters but 4 parameters were supplied
  --> chain/tests/chain_test_helper.rs:63:15
   |
63 |     let reward = reward::output(&keychain, &key_id, 0, false).unwrap();
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 5 parameters

error[E0061]: this function takes 5 parameters but 4 parameters were supplied
  --> chain/tests/chain_test_helper.rs:92:16
   |
92 |         let reward = libtx::reward::output(keychain, &pk, 0, false).unwrap();
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 5 parameters
```

**Comment by quentinlesceller** [comment]
Hi @cadmuspeverell, could you be able to make these changes? However I can copy your branch and do it.

**Comment by quentinlesceller** [comment]
Closed for https://github.com/mimblewimble/grin/pull/2951.

**Pull request closed without merging**